### ISSUE TITLE
Créer un connecteur CAS

### ISSUE THREAD

**Issue by prytoegrian** [libertempo/api#93]
Cf. #51 

Nous nous aiderons pour cela de https://github.com/apereo/phpCAS. 

Nous avions noté ceci dans le fichier de spec : 
> Par nature, ce moyen déporte l’identification d’une personne sur un serveur centralisé, suite à quoi l’utilisateur est redirigé sur l’application initiale avec un message de succès. Puisque l’API ne peut pas rediriger l’utilisateur, il est nécessaire qu’une autre application réalise la première connexion. Nous allons donc laisser la partie de l’authentification initiale côté application front, cette dernière transmettant à l’API le token d’identification CAS ; côté API, on appellera la méthode CAS `/verify` avec le token et le serveur nous dira si le token est OK.

**Comment by wouldsmina** [comment]
Dans un premier temps, il faut obtenir un Proxy Ticket auprès du serveur CAS depuis lt-web. Ce [PR](https://github.com/libertempo/web/pull/814) nécessite d'être validé. Pour cela, il est indispensable d'avoir un serveur apache configuré avec la chaine de certification (SSL) complète est dans l'ordre attendu par le serveur CAS. Bien sur, l’autorité de certification doit existé dans le keystore du CAS. L'utilisation de [cette exemple](https://github.com/apereo/phpCAS/blob/master/docs/examples/example_pgt_storage_file.php) est un bon début pour valider la conformité de la configuration SSL d'apache.

Une fois le Proxy Ticket récupéré, il faudra le transmettre à l'API en guise de mot de passe.

Conclusion, pas de CAS dans Nooma. :confused:

**Comment by prytoegrian** [comment]
Tu as plus bossé que moi sur ce sujet, je te suis.
Les utilisateurs ont une solution de repli ?

**Comment by wouldsmina** [comment]
>Les utilisateurs ont une solution de repli ?

Oui, ils peuvent utiliser le ldap sur lequel s'appuie le serveur CAS, mais ils perdent le côté SSO.

**Comment by prytoegrian** [comment]
Ok

**Comment by wouldsmina** [comment]
Pas si vite ! 😅
